Refactor guides section + promote HTTP over GIT

Wauplin opened the pull request:
Partly started from a [slack discussion](https://huggingface.slack.com/archives/C02V5EA0A95/p1675787943238249) (internal link) (resolves https://github.com/huggingface/huggingface_hub/issues/1334).

In this PR, I:
- reviewed the [./guides/repository](https://moon-ci-docs.huggingface.co/docs/huggingface_hub/pr_1338/en/guides/repository) page to emphasize on the HTTP methods over Git
- refactored the [./guides/overview](https://moon-ci-docs.huggingface.co/docs/huggingface_hub/pr_1338/en/guides/overview) page to make it more appealing (in my opinion)
- moved all "how-to-xxx" pages to the "guides" folder
- renamed how-to pages URLs, especially "how-to-upstream" as ["download"](https://moon-ci-docs.huggingface.co/docs/huggingface_hub/pr_1338/en/guides/download) and "hot-to-upstream" as ["upload"](https://moon-ci-docs.huggingface.co/docs/huggingface_hub/pr_1338/en/guides/upload)
- added a `_redirects.yml` to handle redirections of renamed pages + fixed internal links
- renamed how-to pages in the navbar, making them more "straight to the point". I replaced sentences/questions by keywords (example: "Interact with Discussions and Pull Requests" => ["Community"](https://moon-ci-docs.huggingface.co/docs/huggingface_hub/pr_1338/en/guides/community)). I always found it difficult for the eyes to go through this list as there was quite some "noise". Happy to discuss if someone thinks it's "too minimal" now.
- tried to harmonize titles. Before it was a mix of infinitive ("download files") and -ing form ("sharing model cards"). Harmonized it to infinitive form evewyhere.
- fixed some previously broken links

cc @julien-c @stevhliu @osanseviero @LysandreJik  for feedback
(a bit worried about the redirections but "should be fine", right? I quickly tested and seems to work well)

HuggingFaceDocBuilderDev commented:
The docs for this PR live [here](https://moon-ci-docs.huggingface.co/docs/huggingface_hub/pr_1338). All of your documentation changes will be reflected on that endpoint.

stevhliu commented on docs/source/guides/repository.mdx:
```diff
@@ -0,0 +1,213 @@
+
+<Tip warning={true}>
+
+If you are used to software engineering, working with platforms such as Gitlab/Github/Bitbucket, your first instinct
+might be to use `git` CLI to clone your repo (`git clone`), commit changes (`git add, git commit`) and push them
+(`git push`). This is still valid when using the Huggingface Hub. However, software engineering and machine learning do
+not share the same requirements and workflows. The `huggingface_hub` library has been implemented to address most of the
+ML use cases in a more efficient way. You can read our [Git vs HTTP paradigm](../concepts/git_vs_http) explanation page
+for more details.
```
Briefly clarify what we mean by saying we address ML use cases more efficiently.

```suggestion
If you are used to working with platforms such as Gitlab/Github/Bitbucket, your first instinct
might be to use `git` CLI to clone your repo (`git clone`), commit changes (`git add, git commit`) and push them
(`git push`). This is still valid when using the Hugging Face Hub. However, software engineering and machine learning do
not share the same requirements and workflows. In ML, it may be more efficient to use HTTP, so you don't have to maintain a local folder with massive model weights. You can read our [Git vs HTTP paradigm](../concepts/git_vs_http) explanation page
for more details.
```

stevhliu commented on docs/source/guides/repository.mdx:
```diff
@@ -0,0 +1,213 @@
+If you want to create and manage a repository on the Hub, your machine must be logged in. If you are not, please refer to
+[this section](../quick-start#login). In the rest of this guide, we will assume that your machine is logged in.
+
+## Repo creation and deletion
```
It'd be nice to add a sentence or two after this header to describe what this section is about so users can quickly skim it. In general, its good to not have an empty section header :)

stevhliu commented on docs/source/guides/repository.mdx:
```diff
@@ -0,0 +1,213 @@
+
+### Create a repository
+
+Create an empty repository with [`create_repo`] and give it a name with the `repo_id` parameter. The `repo_id` is your namespace followed by the repository name: `{username_or_org}/{repo_name}`.
```
```suggestion
Create an empty repository with [`create_repo`] and give it a name with the `repo_id` parameter. The `repo_id` is your namespace followed by the repository name: `(username_or_org/repo_name)`.
```

Wauplin replied:
Let's even remove the parenthesis
```suggestion
Create an empty repository with [`create_repo`] and give it a name with the `repo_id` parameter. The `repo_id` is your namespace followed by the repository name: `username_or_org/repo_name`.
```

stevhliu commented on docs/source/guides/repository.mdx:
```diff
@@ -0,0 +1,213 @@
+## Branches and tags
+
+Git repositories often make use of branches to store different versions of a same repository.
+Tags can also be use to flag a specific state of your repository, for example when releasing a version.
```
```suggestion
Tags can also be used to flag a specific state of your repository, for example, when releasing a version.
```

stevhliu commented on docs/source/guides/repository.mdx:
````diff
@@ -0,0 +1,213 @@
+>>> create_branch("bigcode/the-stack", repo_type="dataset", revision="v0.1-release", tag="v0.1.1", tag_message="Bump release version.")
+```
+
+See also [`delete_branch`] and [`delete_tag`].
````
```suggestion
You can use the [`delete_branch`] and [`delete_tag`] functions in the same way to delete a branch or a tag.
```

stevhliu commented on docs/source/guides/repository.mdx:
````diff
@@ -0,0 +1,213 @@
+)
+``` 
+
+## Change repository configuration
````
I'd also add a sentence or two after this empty section header to describe what its about

stevhliu commented on docs/source/guides/repository.mdx:
```diff
@@ -0,0 +1,213 @@
+
+You can rename your repository on the Hub using [`move_repo`]. Using this method, you can also move the repo from a user to
+an organization. When doing so, there are a [few limitations](https://hf.co/docs/hub/repositories-settings#renaming-or-transferring-a-repo)
+that you should be aware of.
```
Just for a quick preview of what a limitation is

```suggestion
that you should be aware of. For example, you can't transfer your repo to another user.
```

stevhliu commented on docs/source/installation.mdx:
```diff
@@ -142,7 +142,7 @@ encounter any undocumented problem by opening [an issue on Github](https://githu
 from the Hub. On Windows, you must activate developer mode or run your script as admin to
 enable symlinks. If they are not activated, the cache-system still works but in an non-optimized
-manner. Please read [this page](./how-to-cache#limitations) for more details.
+manner. Please read [this page](./guides/manage-cache#limitations) for more details.
```
```suggestion
manner. Please read [the cache limitations](./guides/manage-cache#limitations) section for more details.
```

stevhliu commented on docs/source/quick-start.mdx:
```diff
@@ -62,26 +62,34 @@ Once you have your User Access Token, run the following command in your terminal
-You can also provide your token to the functions and methods. This way you don't need to
-store your token anywhere.
-
+It is also possible to login programmatically without asking the user for the token by directly
+passing the token to [`login`] (i.e. `login(token="hf_xxx")`). If you do so, be careful when
+sharing your source code. It is a best practice to load the token from a secure vault instead
+of saving it in plain in your codebase/notebook.
```
You'll need to add a blank line after `<Tip>` so it renders properly. I would also merge this tip with all the ones below it, so you don't have four separate tips. Tips are kind of like side content, but in this case, I think all of this content is quite important!

```suggestion
It is also possible to login programmatically without being prompted to enter your token by directly
passing the token to [`login`] like `login(token="hf_xxx")`. If you do so, be careful when
sharing your source code. It is a best practice to load the token from a secure vault instead
of saving it explicitly in your codebase/notebook.
```

stevhliu commented on docs/source/quick-start.mdx:
```diff
@@ -62,26 +62,34 @@ Once you have your User Access Token, run the following command in your terminal
+is never stored on the machine.
+</Tip>
 
+<Tip>
```
I think it'd be nice to also clarify here that:

Once you are logged in, all requests to the Hub - even methods that don't require authentication - will use your access token by default.

stevhliu commented on docs/source/quick-start.mdx:
````diff
@@ -125,23 +133,24 @@ need to specify:
 ```
 
-To upload more than one file at a time, take a look at this [guide](how-to-upstream)
+To upload more than one file at a time, take a look at this [guide](./guides/upload)
````
```suggestion
To upload more than one file at a time, take a look at the [Upload](./guides/upload) guide
```

julien-c commented on docs/source/_redirects.yml:
```diff
@@ -0,0 +1,10 @@
+how-to-downstream: guides/download
+how-to-inference: guides/inference
+how-to-manage: guides/repository
+how-to-model-cards: guides/modelcards
```
```suggestion
how-to-model-cards: guides/model-cards
```
i would maybe keep the dash here for consistency with other places we use this term

cc @meg-huggingface @EziOzoani

Wauplin replied:
Good idea, I made the change.

EziOzoani replied:
updating to keep a naming consistency sounds good👍

Wauplin commented:
@stevhliu Thank you very much for your extensive review! Looks much nicer now :) I think I've addressed all of your comments.

(cc @julien-c  I also renamed `modelcards.mdx` to `model-cards.mdx` for consistency with the terminology).

osanseviero reviewed (commented):
Very nice! The improvements look very good :fire:

Wauplin commented on docs/source/guides/overview.mdx:
```diff
@@ -3,12 +3,89 @@
-- [Access the Inference API for accelerated inference](../how-to-inference).
-- [Interact with Discussions and Pull Requests ](../how-to-discussions-and-pull-requests).
\ No newline at end of file
+<div class="mt-10">
```
I was about to and then I thought "let's put the same color I use for the guide section in the main page for consistency" i.e. "this type of blue means it's a guide". 
Also a bit lazy to randomize that, especially when more guides will be added.

Wauplin commented on docs/source/guides/repository.mdx:
```diff
@@ -0,0 +1,222 @@
+If you are used to working with platforms such as Gitlab/Github/Bitbucket, your first instinct
+might be to use `git` CLI to clone your repo (`git clone`), commit changes (`git add, git commit`) and push them
+(`git push`). This is still valid when using the Hugging Face Hub. However, software engineering and machine learning do
+not share the same requirements and workflows. In ML, it may be more efficient to use HTTP, so you don't have to maintain
```
Yep completely! I'm updating this :)

Wauplin commented:
It's merged! :)
